build.gradle

```diff
@@ -193,10 +193,26 @@ project('shared:authplugin') {
     }
 }
 
+project('shared:basic-authplugin') {
+    dependencies {
+        compile project(':common')
+        compile project(':shared:authplugin')
+        compile project(':shared:security')
+    }
+
+    javadoc {
+        title = "AuthHandler supporting HTTP Basic Authentication"
+        dependsOn delombok
+        source = delombok.outputDir
+        failOnError = true
+    }
+}
+
 project('shared:security') {
```

Comment:
Just to confirm my understanding of the dependency changes by taking a concrete example of an existing implementation:

https://github.com/pravega/pravega-keycloak/blob/master/build.gradle

Is it correct to say that in the above's:

```
 dependencies {
        compileOnly "org.slf4j:slf4j-api:${slf4jVersion}"
        compileOnly "io.pravega:pravega-shared-authplugin:${pravegaVersion}"
        compileOnly "io.pravega:pravega-client:${pravegaVersion}"
        implementation 'org.keycloak:keycloak-authz-client'
...
```

it would now become:

```
 dependencies {
        compileOnly "org.slf4j:slf4j-api:${slf4jVersion}"
        compileOnly "io.pravega:pravega-shared-security:${pravegaVersion}"
        implementation 'org.keycloak:keycloak-authz-client'
...
```
?

The fact we needed the client and authplugin before was because of this and this respectively:

https://github.com/pravega/pravega-keycloak/blob/master/client/src/main/java/io/pravega/keycloak/client/PravegaKeycloakCredentials.java#L13

https://github.com/pravega/pravega-keycloak/blob/master/client/src/main/java/io/pravega/keycloak/client/PravegaKeycloakCredentials.java#L17

But now both would come from the single shared-security module, is that correct?

Reply:
Yes, that is correct, except that `AuthConstants` which contains constants applicable to auth plugins is still under `"io.pravega:pravega-shared-authplugin`, so you'll need to have a dependency on that module. To remove the dependency, the implementation may choose to use their own String constants instead of the ones in `io.pravega.auth.AuthConstants`.